### PR TITLE
Feature/omc

### PR DESCRIPTION
**Pull request by jbwahl**
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes?
- [x] Have you successfully run the Feel++ testsuite with your changes locally?

-----

### CODE REVIEW

**Comment by jbwahl** [comment]
The new features in FeelFMI framework

File: feel/feelfmi/fmu.cpp [hunk 1]
```diff
@@ -25,6 +25,15 @@ FMU::FMU( std::string prefix ) :
...
     if ( !(boost::filesystem::exists(dir)) )
         boost::filesystem::create_directory(dir);
+
+    M_export_directory = soption( _name="fmu.export-directory", _prefix=M_prefix);
+    if ( M_export_directory!="" )
+        M_export_directory = Environment::expand( M_export_directory );
+
+    if ( Environment::vm().count(prefixvm( M_prefix,"fmu.exported-variables" )) )
+    {
+        M_export_list = var_list_ptrtype( new var_list_type( option( _name="fmu.exported-variables", _prefix=M_prefix).template as<std::vector<std::string> >() ) );
```
Comment (prudhomm) [inline feel/feelfmi/fmu.cpp]: use `make_shared<>`
```
M_export_list = make_shared<var_list_type>( option( _name="fmu.exported-variables", _prefix=M_prefix).template as<std::vector<std::string> >() );
```

File: feel/feelfmi/fmu.cpp [hunk 2]
```diff
@@ -58,13 +67,22 @@ int FMU::load( std::string _path )
...
+    }
 
     return 1;
 }
 
 
 void FMU::simulate( double t_init, double t_final, double tolerance )
 {
+    if ( t_init<0 )
+        t_init = doption( _name="fmu.time-initial", _prefix=M_prefix );
```
Comment (prudhomm) [inline feel/feelfmi/fmu.cpp]: define member variable from a constructor to hold `fmu.time-initial`value. do not use `<>option` functions in a member function.

File: feel/feelfmi/fmu.cpp [hunk 3]
```diff
@@ -58,13 +67,22 @@ int FMU::load( std::string _path )
...
+    if ( t_init<0 )
+        t_final = doption( _name="fmu.time-final", _prefix=M_prefix );
```
Comment (prudhomm) [inline feel/feelfmi/fmu.cpp]: idem as above

File: feel/feelfmi/fmu.cpp [hunk 4]
```diff
@@ -121,6 +139,31 @@ void FMU::printModelInfo()
...
+    {
+        M_export_list->insert( M_export_list->end(), var_list.begin(), var_list.end() );
+        M_model->setExportList( M_export_list );
+        M_model->setExportDirectory( M_export_directory );
+    }
+}
+
+void FMU::setExportedVariables( std::vector<std::string> const& var_list )
+{
+    M_export_list = var_list_ptrtype( new var_list_type( var_list ) );
```
Comment (prudhomm) [inline feel/feelfmi/fmu.cpp]: use `make_shared<>`

see e.g https://www.boost.org/doc/libs/1_66_0/libs/smart_ptr/doc/html/smart_ptr.html#make_shared

**Comment by jbwahl** [comment]
fixed with 254cb4a71db125e2b8ed2aca5d7d369f49935f9f
@prudhomm i guess we can merge

File: feel/feelfmi/fmu.cpp [hunk 5]
```diff
@@ -6,7 +6,9 @@ namespace Feel
 FMU::FMU( std::string prefix ) :
     M_prefix( prefix ),
     M_verbose( boption(_name="fmu.verbose", _prefix=M_prefix) ),
-    M_callbacks( new callbacks_type )
+    M_callbacks( new callbacks_type ),
```
Comment (prudhomm) [inline feel/feelfmi/fmu.cpp]: is this destroyed in the destructor? do you really need to use `new`?

**Comment by prudhomm** [comment]
@jbwahl please edit the PR description and list the issues that got fixed/closed

**Comment by jbwahl** [comment]
my bad, I check the compilation and execution, but i compiler in the wrong directory... sorry i fix it in develop asap

**Pull request merged**